source/agora/consensus/PreImage.d

```diff
@@ -351,14 +352,14 @@ public struct PreImageCycle
 
     ***************************************************************************/
 
-    public Hash populate (scope const ref Scalar secret, bool consume)
-        @safe nothrow
+    public Hash populate (scope const ref Scalar secret,
+        scope const ref Hash enroll_key, bool consume) @safe nothrow
```

Comment:
Adding the `enroll_key` changes quite a bit. When we do see another validator with the same private key, we don't want to override our own pre-images with their own.